game/ui/city/weeklyfundingscreen.cpp

```diff
@@ -0,0 +1,141 @@
+#include "game/ui/city/weeklyfundingscreen.h"
+#include "forms/form.h"
+#include "forms/graphicbutton.h"
+#include "forms/label.h"
+#include "forms/ui.h"
+#include "framework/event.h"
+#include "framework/framework.h"
+#include "framework/keycodes.h"
+#include "game/state/gamestate.h"
+#include "game/state/rules/weeklyrating.h"
+#include "game/state/shared/organisation.h"
+#include "game/ui/city/scorescreen.h"
+
+namespace OpenApoc
+{
+
+WeeklyFundingScreen::WeeklyFundingScreen(sp<GameState> state)
+    : Stage(), menuform(ui().getForm("city/weekly_funding")), state(state)
+{
+	labelCurrentIncome = menuform->findControlTyped<Label>("FUNDING_CURRENT");
+	labelRatingDescription = menuform->findControlTyped<Label>("SENATE_RATING");
+	labelAdjustment = menuform->findControlTyped<Label>("FUNDING_ADJUSTMENT");
+	labelNextWeekIncome = menuform->findControlTyped<Label>("FUNDING_NEW");
+
+	auto buttonOK = menuform->findControlTyped<GraphicButton>("BUTTON_OK");
+	buttonOK->addCallback(FormEventType::ButtonClick,
+	                      [](Event *) { fw().stageQueueCommand({StageCmd::Command::POP}); });
+}
+
+WeeklyFundingScreen::~WeeklyFundingScreen() = default;
+
+void WeeklyFundingScreen::begin()
+{
+	menuform->findControlTyped<Label>("TEXT_FUNDS")->setText(state->getPlayerBalance());
+	menuform->findControlTyped<Label>("TEXT_DATE")->setText(state->gameTime.getLongDateString());
+	menuform->findControlTyped<Label>("TEXT_WEEK")->setText(state->gameTime.getWeekString());
+
+	menuform->findControlTyped<Label>("TITLE")->setText(tr("WEEKLY FUNDING ASSESSMENT"));
+
+	const UString emptyString;
+	UString ratingDescription;
+
+	const auto player = state->getPlayer();
+	const auto government = state->getGovernment();
+	int currentIncome = player->income;
+
+	if (government->isRelatedTo(player) == Organisation::Relation::Hostile)
+	{
+		ratingDescription =
+		    tr("The Senate has declared total hostility to X-COM and there will be no further "
+		       "funding. Furthermore, the Senate will take any steps necessary to destroy the "
+		       "X-COM organization if it refuses to cease operation.");
+
+		currentIncome = 0;
+		labelAdjustment->setText(emptyString);
```

Comment:
Having to keep around an empty string just to clear text boxes feels a little weird, we probably should have a Label::clear() method or something?

Reply:
Removed default text on the form iteself, no longer needed.